packages/react-notifications/src/components/notification-kinds/api-error-message/api-error-message.tsx

```diff
@@ -0,0 +1,171 @@
+import React from 'react';
+import { useIntl, IntlShape } from 'react-intl';
+import has from 'lodash/has';
+import { reportErrorToSentry } from '@commercetools-frontend/sentry';
+import { TAppNotificationApiError } from '@commercetools-frontend/constants';
+import messages from './messages';
+
+const regexInvalidOperationRequiredAttribute = /Required attribute '(.*)' cannot be removed/;
+
+type ExtraErrorFields = {
+  errorByExtension?: {
+    id: string;
+    key?: string;
+  };
+  localizedMessage?: { [locale: string]: string };
+  field?: string;
+  duplicateValue?: string;
+  invalidValue?:
+    | string
+    | { overlappingPrices: string }
+    | {
+        validFrom: string;
+        validUntil: string;
+      };
+  detailedErrorMessage?: string;
+  attribute?: {
+    name: string;
+  };
+};
+type Props = {
+  error: TAppNotificationApiError<ExtraErrorFields>;
+};
+
+const ApiErrorMessage = (props: Props) => {
+  const intl = useIntl();
+
+  // Attempt to map the error to a specific error message
+  const formattedMessage = getMessageByErrorCode(props.error, intl);
+  if (formattedMessage) {
+    return <>{formattedMessage}</>;
```

Comment:
The fragment is necessary to tell react that this returns a JSX element, otherwise it's a string